sunpy/util/sphinx/minigallery.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import sphinx.util.docutils as sphinx_docutils
+from docutils import statemachine
+from docutils.io import FileInput
+from docutils.parsers.rst import Directive
+
+__all__ = ['MiniGallery']
+
+
+class MiniGallery(Directive):
+    """
+    Custom directive to insert a mini-gallery
+
+    The required argument is the qualified name of the object.  The mini-gallery will be the subset
+    of gallery examples that make use of that object (from that specific namespace).  There can be
+    more than one object named, separated by spaces.
+    """
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+
+    def run(self):
+        # Respect the same disabling options as the ``raw`` directive
+        if (not self.state.document.settings.raw_enabled
+                or not self.state.document.settings.file_insertion_enabled):
+            raise self.warning('"%s" directive disabled.' % self.name)
```

Comment:
```suggestion
            raise self.warning(f'"{self.name}" directive disabled.')
```

Reply:
I am going to leave this because we are going to remove it soon as it's been upstreamed into sphinx-gallery